src/main/java/uk/gov/justice/hmpps/datacompliance/web/controllers/RetentionController.java

```diff
@@ -34,16 +45,42 @@
             nickname = "getRetentionRecord" )
     @ApiResponses(value = {
             @ApiResponse(code = 200, message = "OK", response = String.class),
-            @ApiResponse(code = 404, message = "Requested resource not found.", response = ErrorResponse.class),
-            @ApiResponse(code = 500, message = "Unrecoverable error occurred whilst processing request.", response = ErrorResponse.class),
+            @ApiResponse(code = 404, message = "Requested resource not found", response = ErrorResponse.class),
+            @ApiResponse(code = 500, message = "Unrecoverable error occurred whilst processing request", response = ErrorResponse.class),
     })
     @GetMapping(path = "/offenders/{offenderNo}")
-    public ManualRetention getRetentionRecord(
+    public ResponseEntity<ManualRetention> getRetentionRecord(
             @ApiParam(value = "offenderNo", required = true, example = "A1234BC")
             @NotEmpty
             @PathVariable("offenderNo") final String offenderNo) {
         return retentionService.findManualOffenderRetention(new OffenderNumber(offenderNo))
+                .map(body -> ok().eTag(retentionService.getETag(body)).body(transform(body)))
                 .orElseThrow(EntityNotFoundException::new);
     }
 
+    @ApiOperation(
+            value = "Update a retention record",
+            notes = "Update the reasons why an offender record should be kept beyond the standard period. If a record does not yet exist, a new one will be created.",
+            nickname = "updateRetentionRecord" )
+    @ApiResponses(value = {
+            @ApiResponse(code = 201, message = "Created", response = String.class),
+            @ApiResponse(code = 204, message = "No content", response = String.class),
+            @ApiResponse(code = 400, message = "Invalid request", response = ErrorResponse.class),
+            @ApiResponse(code = 500, message = "Unrecoverable error occurred whilst processing request", response = ErrorResponse.class),
+    })
+    @PutMapping(path = "/offenders/{offenderNo}")
+    public ResponseEntity<?> updateRetentionRecord(final WebRequest webRequest,
+                                                   @ApiParam(value = "offenderNo", required = true, example = "A1234BC")
+                                                   @NotEmpty @PathVariable("offenderNo") final String offenderNo,
+                                                   @RequestBody ManualRetentionRequest manualRetentionRequest) {
+
+        return retentionService.updateManualOffenderRetention(
+                new OffenderNumber(offenderNo),
+                manualRetentionRequest,
+                webRequest.getHeader("If-Match"))
+
+                .map(ignored -> noContent().build())
+
+                .orElse(created(URI.create("/retention/offenders/" + offenderNo)).build());
```

Comment:
See this a lot and whilst this isn't a problem here we don't really use the (lazy) supplier option `.orElseGet(()-> ...)`.  Just wonder if there a good reason not to???  You can get burnt by this...

Reply:
Yes, very true, I wasn't particularly concerned here since it was just an inexpensive object, but good point!